packages/strapi/lib/core-api/service.js

```diff
@@ -9,14 +9,42 @@ const {
   },
 } = require('strapi-utils');
 
-const getFetchParams = params => {
+/**
+ * Default limit values from config
+ * @return {{maxLimit: number, defaultLimit: number}}
+ */
+const getLimitConfigDefaults = () => ({
+  defaultLimit: _.toNumber(strapi.config.get('api.rest.defaultLimit', 100)),
+  maxLimit: _.toNumber(strapi.config.get('api.rest.maxLimit', -1)), // (-1) unlimited by default
+});
+
+const getLimitParam = params => {
+  const { defaultLimit, maxLimit } = getLimitConfigDefaults();
+  if (!params._limit) {
```

Comment:
even if it is an edge case if limit is 0 it will return the default but 0 is a valid limit :)